org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/InitHandler.java

```diff
@@ -163,6 +163,16 @@ InitializeResult initialize(InitializeParams param) {
 		if (!preferenceManager.getClientPreferences().isDocumentHighlightDynamicRegistered()) {
 			capabilities.setDocumentHighlightProvider(Boolean.TRUE);
 		}
+		TextDocumentSyncOptions textDocumentSyncOptions = new TextDocumentSyncOptions();
+		textDocumentSyncOptions.setChange(TextDocumentSyncKind.Incremental);
+		if (preferenceManager.getClientPreferences().isWillSaveRegistered()) {
+			textDocumentSyncOptions.setWillSave(Boolean.TRUE);
+		}
+
+		if (preferenceManager.getClientPreferences().isWillSaveWaitUntilRegistered()) {
+			textDocumentSyncOptions.setWillSaveWaitUntil(Boolean.TRUE);
+		}
+		capabilities.setTextDocumentSync(textDocumentSyncOptions);
```

Comment:
something's wrong, with this PR applied, we completely lose autocompletion

Reply:
It maybe caused by the default value of TextDocumentSyncOptions' saveOption and openClose properties are null. 

I get autocompletion work by set them to `true`. 

But during test, I still get error notifications with 'Request textDocument/completion' failed.

P.S My Platform is Archlinux, VSCode 1.19.2